plugins/modules/api.py

```diff
@@ -304,6 +308,11 @@ def __init__(self):
             update=dict(type='str'),
             cmd=dict(type='str'),
             query=dict(type='str'),
+            extended_query=dict(type='dict', options=dict(
+                attributes=dict(type='list', required=True),
```

Comment:
This way Ansible makes sure that every element of this list is a string:
```suggestion
                attributes=dict(type='list', elements='str', required=True),
```